scripts/generate_forger_keys.sh

```diff
@@ -25,31 +25,38 @@ done
 if [ -z "${scnode_wallet_seed}" ]; then
-  fn_die "The stack was not yet initialized. Run init.sh script first. There is nothing to start.  Exiting ..."
+  fn_die "The stack was not yet initialized. The wallet seed has not been populated correctly. Please run init.sh script or check your  "${ROOT_DIR}/${ENV_FILE}" file. Exiting ..."
 fi
 
-# Generate Vrf Key (forger keys)
-VRF_KEY="$($COMPOSE_CMD -f ${compose_file} exec "${CONTAINER_NAME}" gosu user curl -s -X POST "http://127.0.0.1:${SCNODE_REST_PORT}/wallet/createVrfSecret" -H "accept: application/json" -H 'Content-Type: application/json' | jq .result[].publicKey)" 
+# Checking if the evm node is running 
+if [ -n "$(docker ps -q -f status=running -f name="${CONTAINER_NAME}")" ]; then
+  echo "Checking if the node is reachable..."
+  scnode_start_check
 
-echo "Generated Vrf public key : ${VRF_KEY}"
-sleep 1 
+  # Generate Vrf Key (forger keys)
+  vrf_key="$($COMPOSE_CMD -f ${compose_file} exec "${CONTAINER_NAME}" gosu user curl -s -X POST "http://127.0.0.1:${SCNODE_REST_PORT}/wallet/createVrfSecret" -H "accept: application/json" -H 'Content-Type: application/json' | jq .result[].publicKey)" 
 
-# Generate blockSignPublicKey (blockSignPublicKey)
-BLOCK_SIGN_KEY="$($COMPOSE_CMD -f ${compose_file} exec "${CONTAINER_NAME}" gosu user curl -s -X POST "http://127.0.0.1:${SCNODE_REST_PORT}/wallet/createPrivateKeySecp256k1" -H "accept: application/json" -H 'Content-Type: application/json' | jq .result[].address)"
+  echo "Generated Vrf public key : ${vrf_key}"
+  sleep 1 
 
-echo "Generated blockSignPublicKey : ${BLOCK_SIGN_KEY}"
-sleep 1
+  # Generate blockSignPublicKey (blockSignPublicKey)
+  block_sign_key="$($COMPOSE_CMD -f ${compose_file} exec "${CONTAINER_NAME}" gosu user curl -s -X POST "http://127.0.0.1:${SCNODE_REST_PORT}/wallet/createPrivateKeySecp256k1" -H "accept: application/json" -H 'Content-Type: application/json' | jq .result[].address)"
 
-#Generate PrivateKeySecp256k1 (Ethereum compatible address key pair)
-ETH_ADDRESS="$($COMPOSE_CMD -f ${compose_file} exec "${CONTAINER_NAME}" gosu user curl -s -X POST "http://127.0.0.1:${SCNODE_REST_PORT}/wallet/createPrivateKey25519" -H "accept: application/json" -H 'Content-Type: application/json' | jq .result[].publicKey)"
+  echo "Generated blockSignPublicKey : ${block_sign_key}"
+  sleep 1
 
-echo "Generated Ethereum address : ${ETH_ADDRESS}"
+  #Generate PrivateKeySecp256k1 (Ethereum compatible address key pair)
+  eth_address="$($COMPOSE_CMD -f ${compose_file} exec "${CONTAINER_NAME}" gosu user curl -s -X POST "http://127.0.0.1:${SCNODE_REST_PORT}/wallet/createPrivateKey25519" -H "accept: application/json" -H 'Content-Type: application/json' | jq .result[].publicKey)"
 
+  echo "Generated Ethereum address : ${eth_address}"
+else
+  fn_die "=== ${CONTAINER_NAME} node is not running. You need to start ${CONTAINER_NAME} node to be able to generate the keys ==="
```

Comment:
I would say better
"...... Make sure ${CONTAINER_NAME} node is up and running"

> fn_die "Error: ${CONTAINER_NAME} node is not running. Make sure it is up and running in order to be able to generate the keys.  Exiting ..."

Reply:
Approved